### PR TITLE
PaginatedDataTable loading indicator compliance with Material Specification

### PR DESCRIPTION
**Pull request by wer-mathurin**
his PR change the loading indicator of the paginated data table with a linear progress indicator like describe in the ticket (#102861)

List which issues are fixed by this PR. You must list at least one issue.
#102861

## Pre-launch Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 1]
```diff
@@ -80,39 +81,47 @@ class PaginatedDataTable extends StatefulWidget {
...
-       assert(rowsPerPage > 0),
-       assert(() {
-         if (onRowsPerPageChanged != null) {
-           assert(availableRowsPerPage != null && availableRowsPerPage.contains(rowsPerPage));
-         }
-         return true;
-       }()),
-       assert(source != null),
-       assert(!(controller != null && (primary ?? false)),
+  })  : assert(actions == null || (actions != null && header != null)),
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/paginated_data_table.dart:97]: there seems to be an addition space before `:` can you revert the change and also the rest of initializers

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 2]
```diff
@@ -371,7 +361,8 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
     pageTo(((_rowCount - 1) / widget.rowsPerPage).floor() * widget.rowsPerPage);
   }
 
-  bool _isNextPageUnavailable() => !_rowCountApproximate &&
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/paginated_data_table.dart:374]: this should be a regular function instead, the flutter style guide suggest s arrow function should be one liner.

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 3]
```diff
@@ -381,7 +372,8 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
     // TODO(ianh): This whole build function doesn't handle RTL yet.
     assert(debugCheckHasMaterialLocalizations(context));
     final ThemeData themeData = Theme.of(context);
-    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    final MaterialLocalizations localizations =
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/paginated_data_table.dart:375]: revert this? flutter doesn't use format, and sometimes one liner is more readable

**Review by chunhtai — COMMENTED** [review]
Can you revert all of the format changes?

**Comment by Hixie** [comment]
@wer-mathurin Thanks for your contribution! Are you interested in addressing the review comments? It's fine if not, we can save this PR and attach it to the issue so that someone else can pick it up in the future.

**Comment by Hixie** [comment]
@wer-mathurin thank you for your contribution! i'm going to close this PR since it's not making progress but please don't hesitate to file a new one if you are interested in working on this again!
